### PR TITLE
Modernize the Sorting HowTo guide

### PR DESCRIPTION
**Pull request by rhettinger**
Since the sorting howto was originally written, we've added dataclasses, partial, total_ordering, casefold, glossary entries, PEP 8 guidance, etc.  

----
📚 Documentation preview 📚: https://cpython-previews--115479.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/sorting.rst [hunk 1]
```diff
@@ -295,3 +320,24 @@ Odds and Ends
...
+
+* :func:`min` and :func:`max` return the smallest and largest values
+  respectively.  They make a single pass over the input data and
+  require almost no auxiliary memory.
+
+* :func:`heapq.nsmallest` and :func:`heapq.nlargest` functions return
+  the *n* smallest and largest values respectively.  These functions
+  make a single pass over the data keeping only *n* elements in memory
+  at a time.  For values of *n* that are small relative to the number of
+  inputs, these functions make many fewer comparisons than a full sort.
```
Comment (eendebakpt) [inline Doc/howto/sorting.rst]:
```suggestion
  inputs, these functions make much fewer comparisons than a full sort.
```

File: Doc/howto/sorting.rst [hunk 2]
```diff
@@ -295,3 +320,24 @@ Odds and Ends
...
+=============
+
+Some applications require only some of the data to ordered.  The standard
+library provides several tools that do less work than a full sort:
+
+* :func:`min` and :func:`max` return the smallest and largest values
+  respectively.  They make a single pass over the input data and
+  require almost no auxiliary memory.
+
+* :func:`heapq.nsmallest` and :func:`heapq.nlargest` functions return
```
Comment (eendebakpt) [inline Doc/howto/sorting.rst]:
```suggestion
* :func:`heapq.nsmallest` and :func:`heapq.nlargest` return
```
(Similar style as above)

File: Doc/howto/sorting.rst [hunk 3]
```diff
@@ -295,3 +320,24 @@ Odds and Ends
...
+  the *n* smallest and largest values respectively.  These functions
```
Comment (eendebakpt) [inline Doc/howto/sorting.rst]:
```suggestion
  the *n* smallest and largest values, respectively.  These functions
```

File: Doc/howto/sorting.rst [hunk 4]
```diff
@@ -295,3 +320,24 @@ Odds and Ends
     >>> newgrades = {'john': 'F', 'jane':'A', 'dave': 'C'}
     >>> sorted(students, key=newgrades.__getitem__)
     ['jane', 'dave', 'john']
+
+Partial Sorts
+=============
+
+Some applications require only some of the data to ordered.  The standard
```
Comment (eendebakpt) [inline Doc/howto/sorting.rst]:
```suggestion
Some applications require only some of the data to be ordered.  The standard
```

File: Doc/howto/sorting.rst [hunk 5]
```diff
@@ -295,3 +320,24 @@ Odds and Ends
...
+library provides several tools that do less work than a full sort:
+
+* :func:`min` and :func:`max` return the smallest and largest values
```
Comment (eendebakpt) [inline Doc/howto/sorting.rst]:
```suggestion
* :func:`min` and :func:`max` return the smallest and largest values,
```

File: Doc/howto/sorting.rst (hunk 1 above)
Comment (rhettinger) [reply Doc/howto/sorting.rst]: "far" is better

**Pull request merged**